src/views/connectionUI.ts

```diff
@@ -490,7 +489,7 @@ export class ConnectionUI {
      * false otherwise
      */
     public async handleFirewallError(uri: string, profile: IConnectionProfile, ipAddress: string): Promise<boolean> {
-        let azureController = new AzureController(this._context);
+        let azureController = this.connectionManager.azureController;
```

Comment:
There's a bunch of places where we're creating the AzureController it looks like - are those fine? Why did this one need to change?

Reply:
I think those should be changed too, when we declare new azureControllers the credentialStoreInitialized will once again be false which will lead to the same code running again unnecessarily.

Reply:
Are you planning on doing that here? What's the effect if we don't? Is that why we're getting multiple prompts - every time we create a new instance of the azure controller it prompts the user again for access?

Reply:
I have removed all the new azureController creations except for where it is created in connectionManager